llvm/lib/Transforms/Utils/LowerMemIntrinsics.cpp

```diff
@@ -655,8 +670,20 @@ static void createMemMoveLoopKnownSize(Instruction *InsertBefore,
 
   IRBuilder<> PLBuilder(InsertBefore);
 
+  Value *CmpSrcAddr = SrcAddr;
+  Value *CmpDstAddr = DstAddr;
+  if ((SrcAS != DstAS)) {
```

Comment:
Extra parens. Make a helper function for this repeated cast insertion?

Reply:
Addressed in 2f60a54, including the above two suggestions.